src/main/java/life/qbic/projectbrowser/controllers/DataHandler.java

```diff
@@ -2669,6 +2669,50 @@ public String getSecondaryName(Sample samp, String datsetSecName) {
     return secondaryName.replace("__", "_").replaceAll("^_+", "").replaceAll("_+$", "");
   }
 
+  /**
+   * generates informative description from dataset secondary name or the dataset's experiment,
+   * given its ID. should only be used if a dataset does not have an associated sample
+   * 
+   * @param datasetSecondaryName
+   * @param experimentID
+   * @return
+   */
+  public String retrieveDatasetInfoWithoutSample(String datasetSecondaryName, String experimentID) {
+    // dataset name has priority
+    if (datasetSecondaryName != null && !datasetSecondaryName.isEmpty()) {
+      return datasetSecondaryName;
+    }
+    // otherwise generate experiment-type-specific information
+    List<Experiment> experiments = getOpenBisClient().getExperimentById2(experimentID);
+    if (experiments.isEmpty()) {
+      LOG.warn("Experiment with ID " + experimentID
+          + " not found. Returning empty metadata to display.");
+      return "";
+    }
+    Experiment e = experiments.get(0);
+    Map<String, String> props = e.getProperties();
+    String type = e.getExperimentTypeCode();
+    switch (type) {
+      case "Q_NGS_NANOPORE_RUN":
+        String res = props.get("Q_FLOWCELL_BARCODE");
+        if (res == null) {
+          LOG.warn("No flow cell barcode was found for Nanopore Run " + experimentID
+              + ". Returning less informative metadata to display.");
+          res = "unclassified reads";
+        } else {
+          res += " unclassified reads";
+        }
+        return res;
+      default:
+        if (props.get("Q_SECONDARY_NAME") == null) {
+          LOG.warn("No secondary name found for experiment " + experimentID
+              + ". Returning empty metadata to display. You might want to implement a specific case for experiments of type "
+              + type);
+          return "";
+        }
+        return props.get("Q_SECONDARY_NAME");
+    }
```

Comment:
Why use a `switch` if there is only one possible option? I would suggest using an if-condition here.
```suggestion
    if(type.equals("Q_NGS_NANOPORE_RUN") {
        String res = props.get("Q_FLOWCELL_BARCODE");
        if (res == null) {
            LOG.warn("No flow cell barcode was found for Nanopore Run " + experimentID
              + ". Returning less informative metadata to display.");
            res = "unclassified reads";
        } else {
            res += " unclassified reads";
        }
        return res;
    } else {
        if (props.get("Q_SECONDARY_NAME") == null) {
          LOG.warn("No secondary name found for experiment " + experimentID
              + ". Returning empty metadata to display. You might want to implement a specific case for experiments of type "
              + type);
          return "";
        }
        return props.get("Q_SECONDARY_NAME");
    }
```

Reply:
There is an indentation with 2 and with 4 spaces here.

Reply:
using switch because I imagine this is how we would implement further experiment-type-specific cases

Reply:
Do we know that there will be other experiment-type-specific cases? Maybe it makes sense to outsource it into a small method when the time comes instead of keeping a long switch case.